javascript/app-discover/src/state/hooks/useCausalDiscoveryRunner.ts

```diff
@@ -154,70 +156,98 @@ export function useCausalDiscoveryRunner() {
 		[cancelLastDiscoveryResultPromise, setLastDiscoveryResultPromise],
 	)
 
+	const runDiscovery = useCallback(async () => {
+		setErrorMessage(undefined)
+
+		const discoveryPromise = await createDiscoveryPromise(
+			dataset,
+			inModelCausalVariables,
+			causalDiscoveryConstraints,
+			causalDiscoveryAlgorithm,
+			updateProgress,
+			algorithmParams,
+		)
+
+		try {
+			setIsLoading(true)
+			const results = await discoveryPromise.promise!
+
+			// only update if the promise is not canceled
+			if (discoveryPromise.isFinished()) {
+				setCausalDiscoveryResultsState(results)
+				setLoadingState(undefined)
+				setErrorMessage(undefined)
+				setIsLoading(false)
+			}
+		} catch (err) {
+			if (!(err instanceof CanceledPromiseError)) {
+				resetCausalDiscoveryResultsState()
+				setErrorMessage((err as Error).message)
+			}
+			setLoadingState(undefined)
+			setIsLoading(false)
+		}
+
+	}, [
+		dataset,
+		createDiscoveryPromise,
+		inModelCausalVariables,
+		causalDiscoveryAlgorithm,
+		causalDiscoveryConstraints,
+		setLoadingState,
+		setCausalDiscoveryResultsState,
+		updateProgress,
+		algorithmParams,
+		resetCausalDiscoveryResultsState,
+		setErrorMessage,
+		setIsLoading,
+	])
+
+	const stopDiscoveryRun = useCallback(async () => {
+		setAutoRun(false)
+		await cancelLastDiscoveryResultPromise()
+	}, [cancelLastDiscoveryResultPromise, setAutoRun])
+
+	useEffect(() => {
+		return () => {
+			void stopDiscoveryRun()
```

Comment:
Similar to above - should we have a .catch here to log out errors? 

Doesn't this effect depend on the `stepDiscoveryRun` function? Please verify that the react-hooks warnings are clearing (at least for code you've added)

Reply:
I added a catch to log `stopDiscoveryRun` errors

The desired effect is to stop the run on component unmount